macros/models/_project/balancer/balancer_protocol_fee_macro.sql

```diff
@@ -77,6 +77,9 @@ WITH pool_labels AS (
         WHERE l.blockchain = '{{blockchain}}'
         AND l.version = '{{version}}'
         AND s.supply > 0
+        {% if is_incremental() %}
+        AND l.day >= now() - interval '2' day
+        {% endif %}     
```

Comment:
when doing a left join, i find it easier and most accurate to keep all filtering of the table on the left side (`balancer_bpt_supply`) to be within the join conditions, rather than join then filter. i'm not 100% sure that'll resolve your issue called out in description, but worth a try.

it also seems backwards that data is missing when doing the join without an incremental filter at all vs. adding one in 🤔  you'd think the new approach would have higher chance of missing data. maybe it comes back to points above.

Reply:
yeah, makes sense. On commit [851ce02](https://github.com/duneanalytics/spellbook/pull/5684/commits/851ce0246158c0443cc7103ddcda3d2cd44e8430), I've converted the spell from materialized to table and the performance loss doesn't seem too intense, although it's not ideal. So, I made the changes you suggested and the results are looking good so far. 
I say we keep the spell materialized and, if the same issue happens over the next few days, we can maybe change it to a table?

Thanks once again, @jeff-dude 🫡